### ISSUE TITLE
Use new Docker images in kitchen tests to get Systemd activate 

### ISSUE THREAD

**Issue by rshad** [wazuh/wazuh-qa#407]
Hi all!

We need to change the Docker images we use in `Kitchen` tests so we get `Systemd` which fits better with Kitchen tests when checking the status of the service, or if it's enabled or disabled, etc .. . 

**Tasks**

- [x] Identify the proper Docker images.
- [x] Adapt `kitchen.yml` to use ENVIRONMENT variables.
- [x] Adapt `run.sh`
- [x] Adapt the corresponding tests.
- [x] Verify the changes
- - [x] Ubuntu
- - - [x] 16.04
- - - [x] 18.04

- [x] CentOS 7.

- [x] Amazon Linux 2.

Kr,

Rshad

**Comment by rshad** [comment]
Working branch:  [feature-407-docker-images-systemd](https://github.com/wazuh/wazuh-qa/tree/feature-407-docker-images-systemd).

**Comment by rshad** [comment]
Hi all!

Regarding the images to use, we will use the following:
- **Ubuntu**: [jrei/systemd-ubuntu](https://hub.docker.com/r/jrei/systemd-ubuntu)
- **CetnOS**: [centos/systemd
](https://hub.docker.com/r/centos/systemd/).

Using Systemd implies to change the `kitchen.yml` configuration and this is adapted to use the following configuration:

**Ubuntu**
```yml
    driver_config:
      image: jrei/systemd-ubuntu
      platform: ubuntu
      hostname: manager00_ubuntu
      run_command: /sbin/init
      privileged: true
      volume:
        - /sys/fs/cgroup:/sys/fs/cgroup:ro
      provision_command:
        - sed -i 's/UsePAM yes/UsePAM no/g' /etc/ssh/sshd_config
```


**Centos**
```yml
    driver_config:
      image: centos/systemd
      platform: centos
      hostname: manager00_centos
      run_command: /sbin/init
      privileged: true
      volumes:
        - /sys/fs/cgroup:/sys/fs/cgroup:ro
      provision_command:
        - sed -i 's/UsePAM yes/UsePAM no/g' /etc/ssh/sshd_config
```

As you can see, we add a new _provisioning_ command:
```yml
 - sed -i 's/UsePAM yes/UsePAM no/g' /etc/ssh/sshd_config
```

This is needed because without it, the user `kitchen `won't be able to login  into the _Kitchen_ instances, getting the error:
```
Jan 27 15:47:12 manager00_ubuntu sshd[22]: fatal: Access denied for user kitchen by PAM account configuration [preauth]
```

The issue is with PAM stack configuration. Your host is configured with pam_access and the default configuration is not allowing external/SSH access for the new user `kitchen`, even though your keys are set up properly. [StackOverflow Reference](https://unix.stackexchange.com/questions/335215/access-denied-for-a-particular-user-by-pam-account-configuration)

Kr,

Rshad

**Comment by rshad** [comment]
Hi all!

We noted that we could make the testing environment more dynamic by parameterizing it.

For example, for `kitchen.yml` we have the configuration of the same platform for each OS just changing the Docker image name and the Platform name. So we unified the configuration in just two parameterized platforms which depends on the value of environment variables, assigned in the same command line, as:
```bash
IMAGE=keepersecurity/amazonlinux2-systemd PLATFORM=amazonlinux kitchen <action>
```

**`kitchen.yml`**
https://github.com/wazuh/wazuh-qa/blob/6ecbc7a5fcc9770043f382fb6ba2a676c4317ade/kitchen/wazuh-puppet/kitchen/kitchen.yml#L1-L43

---

We also adapted the auxiliary script `run.sh`.

https://github.com/wazuh/wazuh-qa/blob/6ecbc7a5fcc9770043f382fb6ba2a676c4317ade/kitchen/wazuh-puppet/kitchen/run.sh#L1-L40.

---

As we changed the instances names in `kitchen.yml` then we needed to adapt `site.pp` manifest template as follows:
https://github.com/wazuh/wazuh-qa/blob/6ecbc7a5fcc9770043f382fb6ba2a676c4317ade/kitchen/wazuh-puppet/kitchen/manifests/site.pp.template#L1-L9

Kr,

Rshad
